Assets/MixedRealityToolkit.Tests/PlayModeTests/ManipulationHandlerTests.cs

```diff
@@ -251,6 +250,112 @@ public IEnumerator ManipulationHandlerOneHandMove()
 
         }
 
+        /// <summary>
+        /// This tests the one hand far movement while camera (character) is moving around.
+        /// The test will check the offset between object pivot and grab point and make sure we're not drifting
+        /// out of the object on pointer rotation - this test is the same for all objects that won't change 
+        /// their orientation to camera while camera / pointer rotates as this will modify the far interaction grab point
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator ManipulationHandlerOneHandMoveFar()
+        {
+            // set up cube with manipulation handler
+            var testObject = GameObject.CreatePrimitive(PrimitiveType.Cube);
+            testObject.transform.localScale = Vector3.one * 0.2f;
+            Vector3 initialObjectPosition = new Vector3(0f, 0f, 1f);
+            testObject.transform.position = initialObjectPosition;
+            var manipHandler = testObject.AddComponent<ManipulationHandler>();
+            manipHandler.HostTransform = testObject.transform;
+            manipHandler.SmoothingActive = false;
+            manipHandler.ManipulationType = ManipulationHandler.HandMovementType.OneHandedOnly;
+
+            // add near interaction grabbable to be able to grab the cube with the simulated articulated hand
+            testObject.AddComponent<NearInteractionGrabbable>();
+
+            yield return new WaitForFixedUpdate();
+            yield return null;
+
+            const int numCircleSteps = 10;
+            const int numHandSteps = 3;
+
+            Vector3 initialHandPosition = new Vector3(0.04f, -0.18f, 0.3f); // grab point on the lower center part of the cube
+            
+            TestHand hand = new TestHand(Handedness.Right);     
+
+            // do this test for every one hand rotation mode
+            for (ManipulationHandler.RotateInOneHandType type = ManipulationHandler.RotateInOneHandType.MaintainRotationToUser; type <= ManipulationHandler.RotateInOneHandType.RotateAboutGrabPoint; ++type)
```

Comment:
Suggestion: use the Enum reflection to make sure you don't miss any values, in case this enum is changed at some point

`foreach (ManipulationHandler.RotateInOneHandType type in Enum.GetValues(typeof(ManipulationHandler.RotateInOneHandType)))`